api_app/analyzers_manager/file_analyzers/yara_scan.py

```diff
@@ -2,142 +2,328 @@
 # See the file 'LICENSE' for copying permission.
 
 import io
+import json
 import logging
 import os
 import zipfile
-from typing import List, Tuple
+from collections import defaultdict
+from pathlib import PosixPath
+from typing import Dict, List, Optional, Set, Tuple, Union
+from urllib.parse import urlparse
 
 import requests
 import yara
+from cache_memoize import cache_memoize
+from django.conf import settings
 from git import Repo
 
 from api_app.analyzers_manager.classes import FileAnalyzer
 from api_app.analyzers_manager.dataclasses import AnalyzerConfig
 from api_app.exceptions import AnalyzerRunException
+from api_app.models import PluginConfig
+from certego_saas.apps.organization.membership import Membership
+from intel_owl.settings._util import set_permissions
 
 logger = logging.getLogger(__name__)
 
 
+def export_ssh_key(function):
```

Comment:
there is no overall explanation of what this does and why you created a decorator. I already know the reason but please add a docstring or anything similar otherwise future developers would cry over it.